src/visitor.rs

```diff
@@ -131,6 +131,27 @@ impl<'a> FmtVisitor<'a> {
         self.block_indent = self.block_indent.block_indent(self.config);
         self.buffer.push_str("{");
 
+        if self.config.remove_blank_lines_at_start_or_end_of_block() {
+            if let Some(stmt) = b.stmts.first() {
+                let snippet = self.snippet(mk_sp(self.last_pos, stmt.span.lo));
```

Comment:
Could we use a byte pos from inner attributes to construct this snippet and avoid calling `s.trim_left_matches(&[' ', '\t', '\r', '\n'][..]).len()]` below?